test/routes/src/server/routes/t_launchpad.js

```diff
@@ -473,25 +458,58 @@ describe('The Launchpad API endpoint', () => {
           .end(done);
       });
     });
-  });
 
-  describe('complete snap authorization route', () => {
-    beforeEach(() => {
-      const memcachedStub = { cache: {} };
-      memcachedStub.get = (key, callback) => {
-        callback(undefined, memcachedStub.cache[key]);
-      };
-      memcachedStub.set = (key, value, lifetime, callback) => {
-        memcachedStub.cache[key] = value;
-        callback(undefined, true);
-      };
-      setMemcached(memcachedStub);
-    });
+    context('when repository is memcached', () => {
+      const repositoryUrl = 'https://github.com/anaccount/arepo';
+      const snapUrl = `${conf.get('LP_API_URL')}/devel/~test-user/+snap/test-snap`;
+
+      beforeEach(() => {
+        const memcachedStub = { cache: {} };
+        memcachedStub.get = (key, callback) => {
+          callback(undefined, memcachedStub.cache[key]);
+        };
+        memcachedStub.set = (key, value, lifetime, callback) => {
+          memcachedStub.cache[key] = value;
+          callback(undefined, true);
+        };
+        setMemcached(memcachedStub);
+
+        memcachedStub.cache[`url:${repositoryUrl}`] = snapUrl;
+      });
+
+      afterEach(() => {
+        setMemcached(null);
+      });
```

Comment:
This pair is unnecessarily wordy given that the containing `describe` already does almost the same thing.  I think you could just have:

```
beforeEach(() => {
  getMemcached().set(`url:${repositoryUrl}`, snapUrl);
});
```

... and omit `afterEach` entirely.